Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/EamArtifactUtil.java

```diff
@@ -209,11 +197,7 @@ private static CorrelationAttribute getCorrelationAttributeFromBlackboardArtifac
             return null;
         }
 
-        if (null != value) {
-            return new CorrelationAttribute(correlationType, value);
-        } else {
-            return null;
-        }
+        return new CorrelationAttribute(correlationType, value);
```

Comment:
I think we still need that null check/null return. The idea is that we return null if the correlation type doesn't match any data for type of artifact. It's not an error that a keyword hit doesn't have an MD5 hash.

Reply:
If you want to switch to an Optional instead of returning null, that would work too.

Reply:
Optional is new to me.  So the method signature would become like this: private static Optional<CorrelationAttribute> getCorrelationAttributeFromBlackboardArtifact(...) and then clients would call the get() function?

Reply:
I think you do an isPresent() check and then get() if it's true. There's an example in your normalize() method

Reply:
Right, I was mostly asking if it seemed ok to change the method signature and clients of that code?

Reply:
Ah sorry. Yes central repo isn't exported to the public API, so we can change anything we want

Reply:
Cool, I'm happy to do that; I'm just hesitant to go making too much in the way of outward facing changes.